src/app/services/postman-sms/__tests__/postman-sms.service.spec.ts

```diff
@@ -0,0 +1,386 @@
+  //         smsData: {
+  //           form: MOCK_FORM_ID,
+  //           collaboratorEmail: MOCK_RECIPIENT_EMAIL,
+  //           recipientNumber: TWILIO_TEST_NUMBER,
+  //           formAdmin: {
+  //             email: MOCK_ADMIN_EMAIL,
+  //             userId: MOCK_ADMIN_ID,
+  //           },
+  //         },
+  //         smsType: SmsType.BouncedSubmission,
+  //         msgSrvcSid: MOCK_VALID_CONFIG.msgSrvcSid,
+  //         logType: LogType.success,
+  //       })
+  //     })
+
+  //     it('should log failure when sending fails', async () => {
+  //       // Arrange
+  //       const expectedMessage = renderBouncedSubmissionSms(MOCK_FORM_TITLE)
+
+  //       // Act
+  //       const actualResult = await SmsService.sendBouncedSubmissionSms(
+  //         {
+  //           recipient: TWILIO_TEST_NUMBER,
+  //           adminEmail: MOCK_ADMIN_EMAIL,
+  //           adminId: MOCK_ADMIN_ID,
+  //           formId: MOCK_FORM_ID,
+  //           formTitle: MOCK_FORM_TITLE,
+  //           recipientEmail: MOCK_RECIPIENT_EMAIL,
+  //         },
+  //         MOCK_INVALID_CONFIG,
+  //       )
+
+  //       // Assert
+  //       expect(actualResult._unsafeUnwrapErr()).toEqual(new InvalidNumberError())
+  //       expect(twilioFailureSpy).toHaveBeenCalledWith({
+  //         to: TWILIO_TEST_NUMBER,
+  //         body: expectedMessage,
+  //         from: MOCK_INVALID_CONFIG.msgSrvcSid,
+  //         forceDelivery: true,
+  //         statusCallback: expect.stringContaining(MOCK_TWILIO_WEBHOOK_ROUTE),
+  //       })
+  //       expect(smsCountSpy).toHaveBeenCalledWith({
+  //         smsData: {
+  //           form: MOCK_FORM_ID,
+  //           collaboratorEmail: MOCK_RECIPIENT_EMAIL,
+  //           recipientNumber: TWILIO_TEST_NUMBER,
+  //           formAdmin: {
+  //             email: MOCK_ADMIN_EMAIL,
+  //             userId: MOCK_ADMIN_ID,
+  //           },
+  //         },
+  //         smsType: SmsType.BouncedSubmission,
+  //         msgSrvcSid: MOCK_INVALID_CONFIG.msgSrvcSid,
+  //         logType: LogType.failure,
+  //       })
+  //     })
+  //   })
+
+  describe('sendVerificationOtp', () => {
+    let mockOtpData: FormOtpData
+    let testForm: IFormSchema
+
+    beforeEach(async () => {
+      testForm = await FormModel.create({
+        title: 'Test Form',
+        emails: [testUser.email],
+        admin: testUser._id,
+        responseMode: FormResponseMode.Email,
+      })
+
+      mockOtpData = {
+        form: testForm._id,
+        formAdmin: {
+          email: testUser.email,
+          userId: testUser._id,
+        },
+      }
+    })
+
+    it('should return MalformedParametersError error when retrieved otpData is null', async () => {
+      // Arrange
+      // Return null on Form method
+      jest.spyOn(FormModel, 'getOtpData').mockResolvedValueOnce(null)
+      const postmanSendSpy = jest
+        .spyOn(PostmanSmsService, 'sendMopSms')
+        .mockResolvedValueOnce(okAsync(true))
```

Comment:
Don't need `.mockResolvedValueOnce(okAsync(true))` since you do not expect it to be called at L249

Reply:
This is actually to prevent an incorrect error thrown as the assertion, with the correct expectation, will never be called as it failed before hand.